mantidproject/mantid
# Add new options and properties to control a default plot zoom / axes limits

### Description of work

A user (Hamish) wanted a way to set a default axes range / zoom when viewing spectra plots since their full data range is much larger than the data they're interested in.

I have added some options in the settings plots window to control this. They set new properties in the `mantid.user.properties` file.

After opening a plot which has had its axes limits changed by these settings, clicking the home button will return to a full view of the data (so nothing is lost).

Fixes #35666

**Report to:** Hamish

### To test:

You will need to build to pick up the change from the user properties template file.

- Load some data and open a spectra plot
- Note the axes min and max values and pick suitable values to crop to.
- Close the plot
- Open the settings widget
- In the plots page, alter the values under 'Default Axes Zoom'
- Close the setting and replot the spectra, should now be zoomed in.
- Clicking the home button should return to the full view.
- Try some different types of plots, multiple spectra, fitting etc.

Before exiting Mantid, go back to the plot setting and untick the axes limits so that they're set back to none in your properties file.



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
